src/tables.jl

```diff
@@ -115,3 +115,65 @@ end
 
 #     end
 # end
+
+"""
+    populate!(A, table, value)
+
+Populate A with the contents of the `value` column in a provided table.
+The provided table contain columns corresponding to the keys in A and support row iteration.
+"""
+function populate!(A, table, value::Symbol)
+    cols = [value, dimnames(A)...]
+    for r in Tables.rows(table)
+        setkey!(A, (Tables.getcolumn(r, c) for c in cols)...)
+    end
+    return A
+end
+
+"""
+    KeyedArray(table, value, keys...; default=undef)
+    NamedDimsArray(table, value, keys...; default=undef)
+
+Construct a KeyedArray/NamedDimsArray from a `table` supporting column and row.
+`keys` columns are extracted as is, without sorting, and are assumed to uniquely identify
+each `value`. The `default` value is used in cases where no value is identified for a given
+keypair.
+"""
+function KeyedArray(table, value::Symbol, keys::Symbol...; kwargs...)
+   return _construct_from_table(KeyedArray, table, value, keys...; kwargs...)
+end
+
+function NamedDimsArray(table, value::Symbol, keys::Symbol...; kwargs...)
```

Comment:
Hmmm, that's a good point... I guess this signature more closely matches the existing `wrapdims` methods than the `KeyedArray` or `NamedDimsArray` constructors.

Reply:
I might be inclined leave use of UniqueVectors or AcceleratedArrays up to the end user via the `populate!` function. That way it can stay as a test-only dependency.

Reply:
Yes. There's an existing `wrapdims` method that lets you apply some type to all the key vectors. Which you can also do by hand, of course, but here they are being generated for you, so perhaps it's even more convenient. I don't think these should be dependencies.